iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/command_line.inl

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CXX_COMMAND_LINE_INL
+#define IOX_HOOFS_CXX_COMMAND_LINE_INL
+
+namespace iox
+{
+namespace cxx
+{
+namespace internal
+{
+template <typename T>
+inline T addEntry(T& value,
+                  const char shortName,
+                  const CommandLineOptions::name_t& name,
+                  const CommandLineParser::description_t& description,
+                  const ArgumentType argumentType,
+                  T defaultValue,
+                  internal::cmdEntries_t& entries,
+                  internal::cmdAssignments_t& assignments)
+{
+    entries.emplace_back(
+        CommandLineParser::entry_t{shortName,
+                                   name,
+                                   description,
+                                   argumentType,
+                                   {TypeInfo<T>::NAME},
+                                   CommandLineOptions::value_t(TruncateToCapacity, convert::toString(defaultValue))});
+    assignments.emplace_back([&value, &entries, index = entries.size() - 1](CommandLineOptions& options) {
+        auto result = options.get<T>(entries[index].longOption);
+        if (result.has_error())
+        {
+            std::cout << "It seems that the switch value of \"";
+            const bool hasShortOption = (entries[index].shortOption != '\0');
+            const bool hasLongOption = (!entries[index].longOption.empty());
+            if (hasShortOption)
+            {
+                std::cout << "-" << entries[index].shortOption;
+            }
+            if (hasShortOption && hasLongOption)
+            {
+                std::cout << ", ";
+            }
+            if (hasLongOption)
+            {
+                std::cout << "--" << entries[index].longOption;
+            }
+
+            std::cout << "\" is not of type \"" << TypeInfo<T>::NAME << "\"" << std::endl;
+            std::terminate();
+        }
+
+        value = result.value();
+    });
+    return defaultValue;
```

Comment:
Why is the default value always returned?

Reply:
Since this function is used in the struct definition. Let's say you have a struct with
```
struct CommandLineArguments {
  int argumentA = addEntry(...);
};
```

Then we would like to assign to `argumentA` always the default value when it was not set via command line.